common/cache/metricsScopeCache_test.go

```diff
@@ -49,6 +50,8 @@ func TestGetMetricsScope(t *testing.T) {
 		metricsCache.Put(t.domainID, t.scopeID, mockMetricsScope)
 	}
 
+	time.Sleep(11 * time.Second)
```

Comment:
Please don't sleep that long time in unit tests. This plus the test below will cause the entire test file take 20 more seconds. You can consider change the flush duration to a parameter when creating the scope cache, and use a very small duration for test, like 100 milliseconds.

another nit: probably consider move those two tests into a single test suite.